clang/lib/Headers/__clang_hip_libdevice_declares.h

```diff
@@ -14,6 +14,12 @@
 #include "hip/hip_version.h"
 #endif // __has_include("hip/hip_version.h")
 
+#ifdef __SPIRV__
+#define __PRIVATE_AS __attribute__((address_space(0)))
```

Comment:
Never use numbered address spaces. Unconditionally use __attribute__((opencl_private))

Reply:
Acknowledged, but the header was already using numbered address spaces (ROCDL is also using numbered rather than generic), and I really don't want to take on fixing the general badness living here/there. We shouldn't be using explicit ASes at all in what is supposed to be a generic interface, but that's a different kettle of fish. I'd like to keep this simple and land it, if possible, within the backport window as it's a hard break for us at the moment.

Reply:
Using opencl_private is the simpler fix. You are calling code declared and defined in OpenCL, with the private address. Without a specific example of why this does not work out of the box, I insist you just go straight to using it